### ISSUE TITLE
[Question] Use multiple border routing interfaces with ot-br-agent

### ISSUE THREAD

**Issue by caipiblack** [openthread/ot-br-posix#1942]
Hi

**Situation:**

We have a yocto Linux board that can be connected to the "user" network through Ethernet or WiFi (STA mode).

The two interfaces are **not bridged**.

We need to run `ot-br-posix` with the two interfaces as we want it to work with the two interfaces.

Currently otbr-agent is started with `-B eth0` and we would like to make it working also with `mlan0` but currently I don't know how to do it.

When I read the specifications (thread 1.3) it seems possible to use multiple link interfaces but I am really confused, i don't know if we need the **BBR feature for this or not**.

**My research:**

In otbr-agent the `-B` argument is `backbone-ifname` or `Backbone interface`

When we look where this is used: 

```
    , mNcp(mInterfaceName.c_str(), aRadioUrls, mBackboneInterfaceName.c_str(), /* aDryRun */ false, aEnableAutoAttach)
#if OTBR_ENABLE_BACKBONE_ROUTER
    , mBackboneAgent(mNcp, aInterfaceName, aBackboneInterfaceName)
#endif
```

Currently, (in my config) `OTBR_ENABLE_BACKBONE_ROUTER` is disabled by the cmake-option `OTBR_BACKBONE_ROUTER=OFF`, so the parameter is only used here:

```
Ncp::ControllerOpenthread() 
 -> otSysInit() 
  -> platformInit() 
   -> ot::Posix::InfraNetif::Get().Init(aPlatformConfig->mBackboneInterfaceName);
```

NOTES:

1. The `backbone` term is confusing as it can be used when BBR is OFF.
2. In my case with `OTBR_BACKBONE_ROUTER=OFF`, the parameter `Backbone interface` is used to configure an ICMP6 socket that is bind to the specified interface.

**Questions:**

1. Can you confirm that what i am looking for is `Adjacent Infrastructure Link` ?
3. If yes, how to set an `Adjacent Infrastructure Link` in otbr-agent ?
4. If not, is it possible to use multiple interfaces with otbr-agent ? How ?
5. Also, can someone explain what the `Backbone Router` (BBR) feature really provides *more* than classic border routing (BR) ?
- As I understand from the specs, this is when we want to communicate with multiple thread networks using a `backbone
infrastructure link` (a classic network (ethernet/wifi/vpn/...) ?)

**Comment by wgtdkp** [comment]
> 1. Can you confirm that what i am looking for is Adjacent Infrastructure Link ?
Yes, the `Backbone interface` specified by `-B eth0` is the `Adjacent Infrastructure Link`. It's a legacy name.

If you just looking for a feature "run otbr-agent with the first available interface in `set('wlan0', 'eth0')`", you can do this with otbr-agent options: `-B wlan0 -B eth0`

If you want `otbr-agent` to run on both interface at the same, it's not currently supported by neither openthread nor the Thread spec. The spec doesn't support it as well, because the on-link prefixes for the AILs are identical (i.e. derived from the extended pan id) but it doesn't work when Wi-Fi and Ethernet have the same on-link prefixes but not bridged.

**Comment by caipiblack** [comment]
Thanks for the response.

So finally we monitor the WiFi connection events using wpa_cli, then depending on WiFi status we start/restart otbr-agent on the correct interface (mlan0 or eth0)

We considere the WiFi interface as priority interface.

Later we should see how to do it better, for exemple using bridge or bounding.

I close the issue as this is answered.
